### PR TITLE
Add machine-specific docs

### PR DESCRIPTION
**Pull request by mcarlson801**
This PR moves machine-specific scripts for Perlmutter/Frontier/CEE from legacy directory to new landIce/machines directory. It also adds documentation for each of these machines as well as readmes for blake and weaver.

### CODE REVIEW

File: doc/LandIce/machines/cee/do-cmake-trilinos-cee-clang-serial [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+\
+      -D Amesos2_ENABLE_KLU2=ON \
+      -D Intrepid2_ENABLE_TESTS=OFF \
+      -D Phalanx_INDEX_SIZE_TYPE:STRING="INT" \
+      -D Phalanx_KOKKOS_DEVICE_TYPE:STRING="SERIAL" \
+      -D Phalanx_SHOW_DEPRECATED_WARNINGS=OFF \
+      -D Sacado_ENABLE_COMPLEX=OFF \
+      -D Teuchos_ENABLE_COMPLEX=OFF \
+      -D Tpetra_ASSUME_CUDA_AWARE_MPI=OFF \
+      -D Tpetra_INST_INT_LONG_LONG=ON \
```
Comment (mperego) [inline doc/LandIce/machines/cee/do-cmake-trilinos-cee-clang-serial:76]: Can you remove all the lines including `Epetra`? It's not longer needed to explicitly turn them off.
Also can you remove the line for `Amesos`, `AztecOO`, `Ifpack`,`Intrepid`,`ML`,`Pamgen`,`Stokhos`?

File: doc/LandIce/machines/cee/do-cmake-trilinos-cee-gcc-serial [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      -D Phalanx_KOKKOS_DEVICE_TYPE:STRING="SERIAL" \
+      -D Phalanx_SHOW_DEPRECATED_WARNINGS=OFF \
+      -D Sacado_ENABLE_COMPLEX=OFF \
+      -D Teuchos_ENABLE_COMPLEX=OFF \
+      -D Tpetra_ASSUME_CUDA_AWARE_MPI=OFF \
+      -D Tpetra_INST_INT_LONG_LONG=ON \
+      -D Tpetra_INST_INT_INT=OFF \
+      -D MueLu_ENABLE_Tutorial=OFF \
+      -D MueLu_ENABLE_TESTS=OFF \
+\
```
Comment (mperego) [inline doc/LandIce/machines/cee/do-cmake-trilinos-cee-gcc-serial:80]: Can you remove all the lines including `Epetra`? It's not longer needed to explicitly turn them off.
Also can you remove the line for `Amesos`, `AztecOO`, `Ifpack`,`Intrepid`,`ML`,`Pamgen`,`Stokhos`?

File: doc/LandIce/machines/cee/do-cmake-trilinos-cee-intel-serial [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      -D Phalanx_SHOW_DEPRECATED_WARNINGS=OFF \
+      -D Sacado_ENABLE_COMPLEX=OFF \
+      -D Teuchos_ENABLE_COMPLEX=OFF \
+      -D Tpetra_ASSUME_CUDA_AWARE_MPI=OFF \
+      -D Tpetra_INST_INT_LONG_LONG=ON \
+      -D Tpetra_INST_INT_INT=OFF \
+      -D MueLu_ENABLE_Tutorial=OFF \
+      -D MueLu_ENABLE_TESTS=OFF \
+\
+      -D Trilinos_ENABLE_Amesos2=ON \
```
Comment (mperego) [inline doc/LandIce/machines/cee/do-cmake-trilinos-cee-intel-serial:90]: Can you remove all the lines including `Epetra`? It's not longer needed to explicitly turn them off.
Also can you remove the line for `Amesos`, `AztecOO`, `Ifpack`,`Intrepid`,`ML`,`Pamgen`,`Stokhos`?

**Review by mperego — COMMENTED** [review]
Thanks for doing this!! Looks great, but let's simplify a bit the Trilinos cmake configuration file (see comments).

File: doc/LandIce/machines/cee/do-cmake-trilinos-cee-clang-serial (hunk 1 above)
Comment (mcarlson801) [reply doc/LandIce/machines/cee/do-cmake-trilinos-cee-clang-serial:76]: Does this include the line for `EpetraExt`?

Comment (mperego) [reply doc/LandIce/machines/cee/do-cmake-trilinos-cee-clang-serial:76]: Yes. Thanks!

**Review by mperego — APPROVED** [review]

**Pull request merged**